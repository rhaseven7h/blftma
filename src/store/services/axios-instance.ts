import axios from 'axios';

export const axiosInstance = axios.create({
  headers: {
    'Content-Type': 'application/json'
  }
});

export const setupInterceptors = () => {
  axiosInstance.interceptors.request.use(
    (config) => {
      return config;
    },
    (error) => {
      return Promise.reject(error);
    }
  );

  // Add a response interceptor
  axiosInstance.interceptors.response.use(
    (response) => {
      // You can modify the response data here, e.g., handling pagination
      return response;
    },
    (error) => {
      return Promise.reject(error);
    }
  );
};
