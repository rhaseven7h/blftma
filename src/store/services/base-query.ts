import axiosInstance from '@/store/services/axios-instance';
import { BaseQueryArgs, BaseQueryError } from '@/types/base-query';
import { BaseQueryFn } from '@reduxjs/toolkit/query/react';
import { AxiosError, AxiosRequestConfig } from 'axios';

export const axiosBaseQuery = (
  { baseUrl }: { baseUrl: string }
): BaseQueryFn<BaseQueryArgs, unknown, BaseQueryError> => {
  return async ({ url, method, data, params, headers }: AxiosRequestConfig) => {
    try {
      const result = await axiosInstance.request({
        url: baseUrl + url,
        method,
        data,
        params,
        headers
      });
      return { data: result.data };
    } catch (axiosError) {
      const err = axiosError as AxiosError;
      return {
        error: {
          status: err.response?.status,
          data: err.response?.data || err.message
        }
      };
    }
  };
};
