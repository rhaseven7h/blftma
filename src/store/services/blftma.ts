import { Account, AccountsResult, GetAccountsQueryArgs } from '@/types/accounts';
import { createApi, FetchArgs, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

// Define a service using a base URL and expected endpoints
const blftmaApi = createApi({
  reducerPath: 'blftmaApi',
  baseQuery: fetchBaseQuery({ baseUrl: '/api' }),
  tagTypes: ['Account'],
  endpoints: (builder) => ({
    getAccounts: builder.query<AccountsResult, GetAccountsQueryArgs>({
      query: (queryParams: GetAccountsQueryArgs) => {
        return {
          url: `/accounts`,
          params: queryParams,
          method: 'GET'
        } as FetchArgs;
      },
      providesTags: ['Account']
    }),

    createAccount: builder.mutation<Account, Omit<Account, 'id'>>({
      query: (body) => ({
        url: `/accounts`,
        method: 'POST',
        body
      }),
      invalidatesTags: ['Account']
    })
  })
});

export default blftmaApi;
