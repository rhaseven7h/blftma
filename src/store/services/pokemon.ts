import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

type Pokemon = any;

// Define a service using a base URL and expected endpoints
const pokemonApi = createApi({
  reducerPath: 'pokemonApi',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://pokeapi.co/api/v2/' }),
  endpoints: (builder) => ({
    getPokemonByName: builder.query<Pokemon, string>({
      query: (name) => `pokemon/${name}`
    })
  })
});

export default pokemonApi;
