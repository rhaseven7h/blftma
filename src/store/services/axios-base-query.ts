import { axiosInstance } from '@/store/services/axios-instance';
import { BaseQueryFn } from '@reduxjs/toolkit/query/react';
import { AxiosError, AxiosRequestConfig } from 'axios';

export const axiosBaseQuery = ({
  baseUrl
}: {
  baseUrl: string;
}): BaseQueryFn<AxiosRequestConfig, unknown, AxiosError> => {
  return async ({ url, method, data, params, headers }: AxiosRequestConfig) => {
    try {
      const result = await axiosInstance.request({
        url: baseUrl + url,
        method,
        data,
        params,
        headers
      });
      return { data: result.data };
    } catch (axiosError) {
      const err = axiosError as AxiosError;
      return {
        error: err
      };
    }
  };
};
