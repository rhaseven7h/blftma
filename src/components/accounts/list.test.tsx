import AccountsList from '@/components/accounts/list';
import { axiosInstance } from '@/store/services/axios-instance';
import StoreWrapper from '@/util/testing/store-wrapper';
import { render, waitFor } from '@testing-library/react';

const data = {
  'accounts': [
    {
      'id': 8,
      'name': 'Brekke LLC'
    },
    {
      'id': 5,
      'name': 'Donnelly - Kiehn'
    },
    {
      'id': 2,
      'name': 'Fay Inc'
    },
    {
      'id': 4,
      'name': 'Friesen, Kohler and Homenick'
    },
    {
      'id': 6,
      'name': 'Lemke Group'
    }
  ],
  'total': 12,
  'page': 1,
  'size': 5
};

describe('AccountsList', () => {
  it('should render correctly', async () => {
    const requestMock = jest.fn().mockReturnValue({ data });
    jest.spyOn(axiosInstance, 'request').mockImplementation(requestMock);
    const { container, getByTestId } = render(
      <StoreWrapper>
        <AccountsList/>
      </StoreWrapper>
    );
    await waitFor(() => {
      expect(getByTestId('accounts-list-table')).toBeInTheDocument();
      expect(requestMock).toHaveBeenCalledTimes(1);
    });
    expect(container).toMatchSnapshot();
  });
});
