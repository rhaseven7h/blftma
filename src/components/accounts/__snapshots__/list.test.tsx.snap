// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsList should render correctly 1`] = `
<div>
  <h2
    class="text-lg font-bold pb-4"
  >
    accounts
  </h2>
  <div
    class="relative border border-neutral-300 rounded-lg pb-2"
    data-testid="table-element"
  >
    <div
      class="absolute left-0 top-0 -z-10 h-full w-full rounded-lg bg-white drop-shadow-md dark:bg-black"
    />
    <table
      class="w-full text-left text-sm text-gray-500 dark:text-gray-400"
      data-testid="accounts-list-table"
    >
      <thead
        class="group/head uppercase text-gray-700 dark:text-gray-400 text-base"
      >
        <tr>
          <th
            class="px-6 py-3 group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg dark:bg-gray-700 bg-neutral-200"
          >
            account_name
          </th>
          <th
            class="px-6 py-3 group-first/head:first:rounded-tl-lg group-first/head:last:rounded-tr-lg dark:bg-gray-700 bg-neutral-200 text-right"
          >
            actions
          </th>
        </tr>
      </thead>
      <tbody
        class="group/body"
      >
        <tr
          class="group/row odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg"
          >
            Brekke LLC
          </td>
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg flex flex-row flex-nowrap gap-2 justify-end"
          >
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                    />
                    <path
                      d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                    />
                    <path
                      d="M16 5l3 3"
                    />
                  </svg>
                  <span>
                    edit
                  </span>
                </div>
              </span>
            </button>
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 7l16 0"
                    />
                    <path
                      d="M10 11l0 6"
                    />
                    <path
                      d="M14 11l0 6"
                    />
                    <path
                      d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                    />
                    <path
                      d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                    />
                  </svg>
                  <span>
                    delete
                  </span>
                </div>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="group/row odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg"
          >
            Donnelly - Kiehn
          </td>
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg flex flex-row flex-nowrap gap-2 justify-end"
          >
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                    />
                    <path
                      d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                    />
                    <path
                      d="M16 5l3 3"
                    />
                  </svg>
                  <span>
                    edit
                  </span>
                </div>
              </span>
            </button>
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 7l16 0"
                    />
                    <path
                      d="M10 11l0 6"
                    />
                    <path
                      d="M14 11l0 6"
                    />
                    <path
                      d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                    />
                    <path
                      d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                    />
                  </svg>
                  <span>
                    delete
                  </span>
                </div>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="group/row odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg"
          >
            Fay Inc
          </td>
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg flex flex-row flex-nowrap gap-2 justify-end"
          >
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                    />
                    <path
                      d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                    />
                    <path
                      d="M16 5l3 3"
                    />
                  </svg>
                  <span>
                    edit
                  </span>
                </div>
              </span>
            </button>
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 7l16 0"
                    />
                    <path
                      d="M10 11l0 6"
                    />
                    <path
                      d="M14 11l0 6"
                    />
                    <path
                      d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                    />
                    <path
                      d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                    />
                  </svg>
                  <span>
                    delete
                  </span>
                </div>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="group/row odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg"
          >
            Friesen, Kohler and Homenick
          </td>
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg flex flex-row flex-nowrap gap-2 justify-end"
          >
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                    />
                    <path
                      d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                    />
                    <path
                      d="M16 5l3 3"
                    />
                  </svg>
                  <span>
                    edit
                  </span>
                </div>
              </span>
            </button>
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 7l16 0"
                    />
                    <path
                      d="M10 11l0 6"
                    />
                    <path
                      d="M14 11l0 6"
                    />
                    <path
                      d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                    />
                    <path
                      d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                    />
                  </svg>
                  <span>
                    delete
                  </span>
                </div>
              </span>
            </button>
          </td>
        </tr>
        <tr
          class="group/row odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700 hover:bg-gray-50 dark:hover:bg-gray-600"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg"
          >
            Lemke Group
          </td>
          <td
            class="px-6 py-4 group-first/body:group-first/row:first:rounded-tl-lg group-first/body:group-first/row:last:rounded-tr-lg group-last/body:group-last/row:first:rounded-bl-lg group-last/body:group-last/row:last:rounded-br-lg flex flex-row flex-nowrap gap-2 justify-end"
          >
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 7h-1a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-1"
                    />
                    <path
                      d="M20.385 6.585a2.1 2.1 0 0 0 -2.97 -2.97l-8.415 8.385v3h3l8.385 -8.415z"
                    />
                    <path
                      d="M16 5l3 3"
                    />
                  </svg>
                  <span>
                    edit
                  </span>
                </div>
              </span>
            </button>
            <button
              class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-red-700 text-white focus:ring-4 focus:ring-red-300 enabled:hover:bg-red-800 dark:bg-red-600 dark:focus:ring-red-900 dark:enabled:hover:bg-red-700 rounded-lg"
              type="button"
            >
              <span
                class="flex items-stretch transition-all duration-200 rounded-md px-2 py-1 text-xs"
              >
                <div
                  class="flex flex-row flex-nowrap gap-1 items-center"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 7l16 0"
                    />
                    <path
                      d="M10 11l0 6"
                    />
                    <path
                      d="M14 11l0 6"
                    />
                    <path
                      d="M5 7l1 12a2 2 0 0 0 2 2h8a2 2 0 0 0 2 -2l1 -12"
                    />
                    <path
                      d="M9 7v-3a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v3"
                    />
                  </svg>
                  <span>
                    delete
                  </span>
                </div>
              </span>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    class="flex flex-col items-center justify-center"
  >
    <nav
      class=""
    >
      <ul
        class="xs:mt-0 mt-2 inline-flex items-center -space-x-px"
      >
        <li>
          <button
            class="cursor-not-allowed opacity-50 ml-0 rounded-l-lg border border-gray-300 bg-white px-3 py-2 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white inline-flex"
            disabled=""
            type="button"
          >
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
                fill-rule="evenodd"
              />
            </svg>
            Previous
          </button>
        </li>
        <li
          aria-current="page"
        >
          <button
            class="w-12 border border-gray-300 py-2 leading-tight enabled:hover:bg-gray-100 enabled:hover:text-gray-700 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white bg-cyan-50 text-cyan-600 hover:bg-cyan-100 hover:text-cyan-700 dark:border-gray-700 dark:bg-gray-700 dark:text-white"
            type="button"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="w-12 border border-gray-300 bg-white py-2 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
            type="button"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="w-12 border border-gray-300 bg-white py-2 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white"
            type="button"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="rounded-r-lg border border-gray-300 bg-white px-3 py-2 leading-tight text-gray-500 enabled:hover:bg-gray-100 enabled:hover:text-gray-700 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400 enabled:dark:hover:bg-gray-700 enabled:dark:hover:text-white inline-flex"
            type="button"
          >
            Next
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
    <div>
      showing_accounts
       
      <span
        class="font-bold"
      >
        1
      </span>
       
      to
       
      <span
        class="font-bold"
      >
        5
      </span>
       
      out_of
       
      <span
        class="font-bold"
      >
        12
      </span>
       
      accountsLowercase
       
    </div>
    <div
      class="flex flex-row flex-nowrap gap-2 items-center"
    >
      <span>
        in_page
      </span>
      <div
        class="css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="css-13cymwt-control"
        >
          <div
            class="css-hlgwow"
          >
            <div
              class="css-1dimb5e-singleValue"
            >
              1
            </div>
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="true"
              aria-readonly="true"
              class="css-1czm0f-dummyInput-DummyInput"
              id="react-select-2-input"
              inputmode="none"
              role="combobox"
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="css-1wy0on6"
          >
            <span
              class="css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <span>
        out_of
         
        <span
          class="font-bold"
        >
          3
        </span>
         
        pages
      </span>
    </div>
  </div>
</div>
`;
